fiona/_transform.pyx

```diff
@@ -145,10 +144,17 @@ def _transform_geom(
                         <const OGRGeometry *>src_ogr_geom,
                         <OGRCoordinateTransformation *>transform,
                         options)
-        g = _geometry.GeomBuilder().build(dst_ogr_geom)
-
-        _geometry.OGR_G_DestroyGeometry(dst_ogr_geom)
-        _geometry.OGR_G_DestroyGeometry(src_ogr_geom)
+        if dst_ogr_geom == NULL:
+            g = None
+            warnings.warn(
+                "Destination geometry is None. If desired, consider wrapping "
```

Comment:
```suggestion
                "Transformation of the geometry has failed because partial reprojection has not been enabled. To enable partial reprojection wrap "
```